### PR TITLE
FUSETOOLS-2400 & FUSETOOLS-2075 - improve finish action of the new pr…

### PR DESCRIPTION
**Pull request by lhein**
…oject wizard to be more responsable

Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

### CODE REVIEW

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java [hunk 1]
```diff
@@ -111,6 +111,14 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException, Inte
 			}
 		}
 
+		try {
+			Display.getDefault().syncExec( () -> postProjectCreationSetup(prj, subMonitor));
```
Comment (apupier) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java]: you need to call subMonitor.newChild(1) and increase the number from 7 to 8 when creating the submonitor

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java [hunk 2]
```diff
@@ -111,6 +111,14 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException, Inte
...
+		} finally {
+			subMonitor.done();
+		}
+	}
+
+	private void postProjectCreationSetup(IProject prj, SubMonitor subMonitor) {
```
Comment (apupier) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java]: better to use IMonitor as parameter and then call SubMonitor.convert to create a new Submonitor otherwise it means that you would have to track the numbers of operations across function call

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java [hunk 3]
```diff
@@ -142,7 +150,6 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException, Inte
 		// finally open the camel context file
 		openCamelContextFile(prj, subMonitor.newChild(1));
 		new BuildAndRefreshJobWaiterUtil().waitJob(subMonitor);
```
Comment (apupier) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java]: better to call subMonitor.newChild(1) (even if the issue was already there)

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java [hunk 4]
```diff
@@ -308,39 +312,36 @@ private boolean confirmPerspectiveSwitch(IWorkbenchWindow window, IPerspectiveDe
...
-			final IFile[] holder = new IFile[1];
-			searchCamelContextXMLFile(project, holder);
-			try {
-				if (holder[0] == null && project.hasNature(JavaCore.NATURE_ID)) {
-					searchCamelContextJavaFile(project, monitor, holder);
-					isJavaEditorToOpen = true;
-				}
-			} catch (CoreException e1) {
-				ProjectTemplatesActivator.pluginLog().logError(e1);
+		if (project == null) return;
```
Comment (apupier) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java]: I prefer the previous way of writing it, it helps to notice just with the indentation that the code is executed conditionally

File: targetplatform/JBTIS Oxygen Linux x86_64.launch [hunk 5]
```diff
@@ -15,14 +15,14 @@
 <booleanAttribute key="includeOptional" value="true"/>
 <stringAttribute key="location" value="./ws_fusetooling_dev"/>
 <booleanAttribute key="org.eclipse.jdt.launching.ATTR_USE_START_ON_FIRST_THREAD" value="true"/>
-<stringAttribute key="org.eclipse.jdt.launching.JRE_CONTAINER" value="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/java-8-openjdk-amd64"/>
+<stringAttribute key="org.eclipse.jdt.launching.JRE_CONTAINER" value="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/java-1.8.0-openjdk-1.8.0.141-1.b16.fc26.x86_64"/>
```
Comment (apupier) [inline targetplatform/JBTIS Oxygen Linux x86_64.launch]: why this file has been modified?

File: targetplatform/JBTIS Oxygen Mac OS.launch [hunk 6]
```diff
@@ -14,14 +14,14 @@
 <booleanAttribute key="includeOptional" value="true"/>
 <stringAttribute key="location" value="./ws_fusetooling_dev"/>
 <booleanAttribute key="org.eclipse.jdt.launching.ATTR_USE_START_ON_FIRST_THREAD" value="true"/>
-<stringAttribute key="org.eclipse.jdt.launching.JRE_CONTAINER" value="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.launching.macosx.MacOSXType/Java SE 8 [1.8.0_92]"/>
+<stringAttribute key="org.eclipse.jdt.launching.JRE_CONTAINER" value="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/java-1.8.0-openjdk-1.8.0.141-1.b16.fc26.x86_64"/>
```
Comment (apupier) [inline targetplatform/JBTIS Oxygen Mac OS.launch]: why this file has been modified?

File: targetplatform/JBossTools Oxygen Linux x86_64.launch [hunk 7]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<launchConfiguration type="org.eclipse.pde.ui.RuntimeWorkbench">
```
Comment (apupier) [inline targetplatform/JBossTools Oxygen Linux x86_64.launch]: why this file has been created?

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java [hunk 8]
```diff
@@ -111,6 +111,14 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException, Inte
 			}
 		}
 
+		try {
+			Display.getDefault().syncExec( () -> postProjectCreationSetup(prj, subMonitor));
+		} finally {
+			subMonitor.done();
```
Comment (apupier) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java]: no need to call done on submonitor see 4.0 https://eclipse.org/articles/Article-Progress-Monitors/article.html

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java (hunk 4 above)
Comment (lhein) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/FuseIntegrationProjectCreatorRunnable.java]: Yeah, but Sonar complained about a too high complexity of the method.

File: targetplatform/JBTIS Oxygen Linux x86_64.launch (hunk 5 above)
Comment (lhein) [inline targetplatform/JBTIS Oxygen Linux x86_64.launch]: because I renamed it to fit how u named it on Windows and to reflect we are no longer using JBTIS TP

File: targetplatform/JBTIS Oxygen Mac OS.launch (hunk 6 above)
Comment (lhein) [inline targetplatform/JBTIS Oxygen Mac OS.launch]: thats odd. I did not touch that one. Must have been changed automagically :O

File: targetplatform/JBossTools Oxygen Linux x86_64.launch (hunk 7 above)
Comment (lhein) [inline targetplatform/JBossTools Oxygen Linux x86_64.launch]: thats odd. I did not touch that one. Must have been changed automagically :O

File: targetplatform/JBTIS Oxygen Linux x86_64.launch (hunk 5 above)
Comment (apupier) [inline targetplatform/JBTIS Oxygen Linux x86_64.launch]: better to be done in another commit and another PR

**Comment by lhein** [comment]
the GTK3_SWT issue should be fixed by using the fork = true param for invoking the finish job runner. I had the same issue here on Fedora but after setting fork to true it helped to display the progress bar correctly.

File: targetplatform/JBossTools Oxygen Linux x86_64.launch (hunk 7 above)
Comment (lhein) [inline targetplatform/JBossTools Oxygen Linux x86_64.launch]: wait...that one is caused by the rename

File: targetplatform/JBTIS Oxygen Linux x86_64.launch (hunk 5 above)
Comment (lhein) [inline targetplatform/JBTIS Oxygen Linux x86_64.launch]: actually it should have been deleted

**Comment by apupier** [comment]
I'm not really convinced about the "fork", when forking, there are a lot of dialog that are popping up and which let's cancel some jobs which might lead to some errors for project creation.
Is it only on Windows?
I preferred having just the progress bar in the bottom of the wizard.
see the videos:
[Create with Fork](https://drive.google.com/a/redhat.com/file/d/0B1cUBNButvV9dy1fd2RZQXROZ1U/view?usp=sharing)
[Create without fork](https://drive.google.com/open?id=0B1cUBNButvV9a0gxYmFWZFd0ZUk)

**Comment by lhein** [comment]
I am recreating a fresh PR.
